recipes/tree-gen/all/test_package/src/primitives.hpp

```diff
@@ -0,0 +1,90 @@
+/** \file
```

Comment:
OK! I see. These files are actually part of the sources of `test_package`. They are needed, but won't be generated by `tree-gen`. Great spot!

Reply:
As far as I see, it's an example from the project, but it requires extra files from there to work. Honestly, we try to try full example as test package because they are bigger than needed to validate a test package. We usually only print a version() method from library and that's it. The tree-gen is a bit more complex because have a code generator involved with cmake.

Reply:
That's it, for this case we want to test the `tree-gen` executable as well as the `tree-gen` library.